tsdb/engine/tsm1/engine.go

```diff
@@ -886,19 +886,53 @@ func (e *Engine) LoadMetadataIndex(shardID uint64, index tsdb.Index) error {
 	return nil
 }
 
-// IsIdle returns true if the cache is empty, there are no running compactions and the
-// shard is fully compacted.
-func (e *Engine) IsIdle() bool {
+func (e *Engine) IsIdle(isLogged bool) bool {
+	return e.loggedIsIdle(isLogged)
+}
+
+const logMsg = "IsIdle false because nonzero"
+
+// loggedIsIdle returns true if the cache is empty, there are no running compactions and the
+// shard is fully compacted.  If trace logging is enabled, it logs the reasons why an engine
+// is busy
+func (e *Engine) loggedIsIdle(isLogged bool) bool {
+	const cacheCompactions = "Cache Compactions: "
+	const levelZeroCompactions = "Level Zero Compactions: "
+	const levelOneCompactions = "Level One Compactions: "
+	const levelTwoCompactions = "Level Two Compactions: "
+	const fullCompactions = "Full Compactions: "
+	const optimizeCompactions = "TSM Optimization Compactions: "
```

Comment:
Could/should these be declared outside of the `func`?

Reply:
If not, is there a benefit to having them as `const`s here vs. inlining them at the callsites below?

Reply:
They could be outside of the func, but I wanted to limit their scope to the minimum necessary.  I liked them as constants simply for line length and the additional clarity of the names.